canonical-web-and-design/maas-ui
# Add betterer test to count progress towards migrating to ts.

## Done
- Add betterer test to count progress towards migrating to ts.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- run `yarn betterer`.
- `cat .betterer.results` and ensure value for "migrate js files to ts" test matches the result of `find . -name "*.js" | wc -l` in the shell.
- rename a `.js` file to `.ts`
- rerun `yarn betterer` - it should say the "migrate" test improved.

## Fixes

Fixes: canonical-web-and-design/MAAS-design#918
